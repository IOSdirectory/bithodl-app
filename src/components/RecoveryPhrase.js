import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';
import RecoveryPhraseWord from './RecoveryPhraseWord';

const styles = StyleSheet.create({
  list: {
    flexWrap: 'wrap',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  }
});

export default class RecoveryPhrase extends Component {
  _renderWords(words) {
    return words.map((word, index) => {
      return (
        <RecoveryPhraseWord key={index} index={index} word={word} />
      );
    });
  }

  render() {
    const phrase = this.props.phrase;
    const words = phrase.split(' ');

    return (
      <View style={[styles.list, this.props.style]}>
        {this._renderWords(words)}
      </View>
    );
  }
}

RecoveryPhrase.propTypes = {
  phrase: PropTypes.string.isRequired,
  style: PropTypes.any
};
