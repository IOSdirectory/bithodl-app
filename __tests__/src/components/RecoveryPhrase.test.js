import React from 'react';
import renderer from 'react-test-renderer';
import RecoveryPhrase from '../../../src/components/RecoveryPhrase';

describe('RecoveryPhrase', () => {
  it('renders correctly', () => {
    const phrase = 'during bulb nominee acquire paddle next course stable govern eagle title wing';

    const tree = renderer.create(
      <RecoveryPhrase phrase={phrase} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
