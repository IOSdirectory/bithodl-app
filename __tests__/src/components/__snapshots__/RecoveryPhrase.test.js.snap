// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecoveryPhrase renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fbfbfb",
          "borderColor": "#DDDDDF",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 7,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
            },
          ]
        }
      >
        1
        . 
        during
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fbfbfb",
          "borderColor": "#DDDDDF",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 7,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
            },
          ]
        }
      >
        2
        . 
        bulb
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fbfbfb",
          "borderColor": "#DDDDDF",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 7,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
            },
          ]
        }
      >
        3
        . 
        nominee
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fbfbfb",
          "borderColor": "#DDDDDF",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 7,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
            },
          ]
        }
      >
        4
        . 
        acquire
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fbfbfb",
          "borderColor": "#DDDDDF",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 7,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
            },
          ]
        }
      >
        5
        . 
        paddle
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fbfbfb",
          "borderColor": "#DDDDDF",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 7,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
            },
          ]
        }
      >
        6
        . 
        next
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fbfbfb",
          "borderColor": "#DDDDDF",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 7,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
            },
          ]
        }
      >
        7
        . 
        course
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fbfbfb",
          "borderColor": "#DDDDDF",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 7,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
            },
          ]
        }
      >
        8
        . 
        stable
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fbfbfb",
          "borderColor": "#DDDDDF",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 7,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
            },
          ]
        }
      >
        9
        . 
        govern
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fbfbfb",
          "borderColor": "#DDDDDF",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 7,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
            },
          ]
        }
      >
        10
        . 
        eagle
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fbfbfb",
          "borderColor": "#DDDDDF",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 7,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
            },
          ]
        }
      >
        11
        . 
        title
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 4,
        "width": "49%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fbfbfb",
          "borderColor": "#DDDDDF",
          "borderRadius": 30,
          "borderWidth": 1,
          "padding": 7,
          "paddingLeft": 12,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        numberOfLines={undefined}
        style={
          Array [
            Object {
              "color": "#504F63",
              "fontFamily": "Avenir Next",
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "Menlo-Regular",
              "fontSize": 11,
            },
          ]
        }
      >
        12
        . 
        wing
      </Text>
    </View>
  </View>
</View>
`;
